internal/server/httpapi/trigger.go

```diff
@@ -0,0 +1,380 @@
+package httpapi
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/gorilla/mux"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/hashicorp/waypoint/internal/clicontext"
+	"github.com/hashicorp/waypoint/internal/serverclient"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+	"github.com/hashicorp/waypoint/pkg/serverconfig"
+)
+
+// Message is the message we return to the requester when streaming output
+type Message struct {
+	JobId    string `json:"jobId,omitempty"`
+	Message  string `json:"message,omitempty"`
+	ExitCode string `json:"exitCode,omitempty"`
+}
```

Comment:
Going to update this `Message` to have a `Value interface{}` and a `ValueType string`